atomic_reactor/source.py

```diff
@@ -137,35 +142,22 @@ def get_vcs_info(self):
 
 
 class GitSource(Source):
-    def __init__(self, provider, uri, dockerfile_path=None, provider_params=None, workdir=None):
-        super(GitSource, self).__init__(provider, uri, dockerfile_path,
-                                        provider_params, workdir)
-        self.git_commit = self.provider_params.get('git_commit', None)
-        branch = self.provider_params.get('git_branch', None)
-        depth = self.provider_params.get('git_commit_depth', None)
-        self.lg = util.LazyGit(self.uri, self.git_commit, self.source_path, branch=branch,
-                               depth=depth)
-
     @property
     def commit_id(self):
-        return self.lg.commit_id
-
-    def get(self):
-        return self.lg.clone()
-
-    @property
-    def path(self):
-        return self.lg.git_path
-
-    def get_vcs_info(self):
-        return VcsInfo(
-            vcs_type='git',
-            vcs_url=self.lg.git_url,
-            vcs_ref=self.lg.commit_id
-        )
-
-    def reset(self, git_reference):
-        self.lg.reset(git_reference)
+        return get_commit_id(self.path)
```

Comment:
is it intentional that this will raise error when the repo isn't cloned ?

Reply:
you could also call self.get() and that would solve it

Reply:
Looking into it some more, I think there are a few options:

1. make `Source.path` always call get()
2. for methods that expect the path to already exist (commit_id, get_vcs_info(), config, figure_out_build_file_path()), raise an error if the path doesn't exist
3. only document that many methods expect the path to exist

I'll go with 2 for now, LMK what you think